### PR TITLE
Minor improvements

### PR DESCRIPTION
**Pull request by Samsinite**
@jweakley and @grelement please review.

### CODE REVIEW

**Comment by grelement** [comment]
:thumbsup:

File: lib/form_me/field.rb [hunk 1]
```diff
@@ -15,10 +20,29 @@ class Field < ActiveRecord::Base
               }
     validates :field_type, inclusion: { in: %w(text select checkbox-array price integer decimal textarea date date-restricted time date-range time-range year link) }
```
Comment (jweakley) [inline lib/form_me/field.rb]: Line length is a little long.

File: lib/form_me/field.rb [hunk 2]
```diff
@@ -15,10 +20,29 @@ class Field < ActiveRecord::Base
...
     validates :label, presence: true
+    validates :uuid, presence: true
     validate :uniqness_of_key_in_related_forms
 
     scope :only_required, -> { where(required: true) }
 
+    def set_key_if_empty
+      unless key
```
Comment (jweakley) [inline lib/form_me/field.rb]: Use a guard clause: http://www.rubydoc.info/github/bbatsov/rubocop/Rubocop/Cop/Style/GuardClause

File: lib/form_me/field.rb [hunk 3]
```diff
@@ -15,10 +20,29 @@ class Field < ActiveRecord::Base
...
+        self.key = label.gsub(/[^0-9a-z ]/i, '').downcase.tr(' ', '_').camelize(:lower)
+        while form.does_field_key_exist?(self)
+          counter += 1
+          self.key = "#{key}#{counter}"
+        end
+      end
+    end
+
+    def set_uuid_if_empty
+      unless uuid
```
Comment (jweakley) [inline lib/form_me/field.rb]: Use a guard clause: http://www.rubydoc.info/github/bbatsov/rubocop/Rubocop/Cop/Style/GuardClause

File: lib/form_me/form.rb [hunk 4]
```diff
@@ -15,21 +15,47 @@ class Form < ActiveRecord::Base
...
                 with: /\A[a-zA-Z0-9_\-]+\z/,
                 message: 'field names can only contain letters and numbers'
               }
 
+    validates :uuid, presence: true
     validate :uniqness_of_key_in_related_forms
 
-    def uniqness_of_key_in_related_forms
+    def set_key_if_empty
+      unless key
```
Comment (jweakley) [inline lib/form_me/form.rb]: Use a guard clause: http://www.rubydoc.info/github/bbatsov/rubocop/Rubocop/Cop/Style/GuardClause

File: lib/form_me/form.rb [hunk 5]
```diff
@@ -15,21 +15,47 @@ class Form < ActiveRecord::Base
...
-          errors.add(:key, 'must be unique among neighbor forms and fields')
-          break
+          return false
         end
       end
+      true
+    end
+
+    def uniqness_of_key_in_related_forms
+      unless is_key_unique_in_related_forms
```
Comment (jweakley) [inline lib/form_me/form.rb]: Use a guard clause: http://www.rubydoc.info/github/bbatsov/rubocop/Rubocop/Cop/Style/GuardClause

**Comment by jweakley** [comment]
:+1: once those style changes are made.

**Comment by Samsinite** [comment]
Issues addressed, going to rebase and merge

**Pull request merged**